helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -35,8 +36,48 @@
 
   void unsubscribeDataChanges(String path, IZkDataListener listener);
 
+  /*
+   * This is for backwards compatibility.
+   *
+   * TODO: remove below default implementation when getting rid of I0Itec in the new zk client.
+   */
+  default void subscribeStateChanges(
+      final org.apache.helix.manager.zk.zookeeper.IZkStateListener listener) {
```

Comment:
Nit: is there a way to switch the way express IZkStateListeners?

So that listener's type here would be just `IZkStateListener` and the method signature in 69 would be the IOItec's `IZkStateListener`.

I am suggesting this because HelixZkClient is in the manager module.

Reply:
Same as: [my reply](https://github.com/apache/helix/pull/644/files#r354952718)